### ISSUE TITLE
RateLimit middleware always uses ipStrategy sourceCriterion

### ISSUE THREAD

**Issue by jfrancisco0** [containous/traefik#6687]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
Configure the RateLimit middleware for an endpoint using the `requestHeaderName` sourceCriterion.

### What did you expect to see?
Requests fall into different buckets for different header values of the header key defined in `sourceCriterion.requestHeaderName`

### What did you see instead?
The `sourceCriterion.requestHeaderName` is ignored, and the defined header has no effect. Even though the middleware appears with the appropriate configuration on the Traefik dashboard, the logs confirm it is using the default sourceCriterion (`ipStrategy`) instead.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Traefik version 2.2.0 built on 2020-03-25T17:32:57Z
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```
Docker labels
- "traefik.enable=true"
- "traefik.http.routers.whoami.rule=Host(`localhost`)"
- "traefik.http.routers.whoami.entrypoints=web"
- "traefik.http.middlewares.test-ratelimit.ratelimit.average=2"
- "traefik.http.middlewares.test-ratelimit.ratelimit.burst=5"
- "traefik.http.middlewares.test-ratelimit.ratelimit.period=1m"
- "traefik.http.routers.whoami.middlewares=test-ratelimit@docker"
- "traefik.http.middlewares.test-ratelimit.ratelimit.sourcecriterion.requestheadername=username"
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
traefik    | time="2020-04-16T12:35:05Z" level=debug msg="Using IPStrategy" middlewareType=RateLimiterType middlewareName=test-ratelimit@docker entryPointName=web routerName=whoami@docker
```

The condition of the if clause in [extractor.go](https://github.com/containous/traefik/blob/v2.2/pkg/middlewares/extractor.go#L26) is always evaluated as `true` due to the [defaults](https://github.com/containous/traefik/blob/v2.2/pkg/config/dynamic/middlewares.go#L341) of this middleware.

At first I was going to suggest changing the order of the `if`'s in [extractor.go](https://github.com/containous/traefik/blob/v2.2/pkg/middlewares/extractor.go#L26) to place `if sourceMatcher.IPStrategy != nil` in the last position, since that is the default, but I think this would break the `InFlightReq` middleware, which has `requestHost` as its default.

**Comment by mpl** [comment]
@jfrancisco0 Please provide your full environment & configuration as requested in the template.

**Comment by jfrancisco0** [comment]
@mpl I am using the following `docker-compose.yaml`:
```version: "3.3"

services:

  traefik:
    image: "traefik:v2.2"
    container_name: "traefik"
    command:
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
      - "--log.level=DEBUG"
      - "--accesslog=true"
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"

  whoami:
    image: "containous/whoami"
    container_name: "simple-service"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.whoami.rule=Host(`localhost`)"
      - "traefik.http.routers.whoami.entrypoints=web"
      - "traefik.http.middlewares.test-ratelimit.ratelimit.average=2"
      - "traefik.http.middlewares.test-ratelimit.ratelimit.burst=5"
      - "traefik.http.middlewares.test-ratelimit.ratelimit.period=1m"
      - "traefik.http.routers.whoami.middlewares=test-ratelimit@docker"
      - "traefik.http.middlewares.test-ratelimit.ratelimit.sourcecriterion.requestheadername=username"```

**Comment by traefiker** [comment]
Closed by #6713.
